debezium-core/src/main/java/io/debezium/rest/MetricsResource.java

```diff
@@ -0,0 +1,132 @@
+                                           String[] namespaces)
+            throws MalformedObjectNameException {
+        List<MetricsDescriptor.Task> tasksPayload = new ArrayList<>();
+        MetricsDescriptor.Connector connectorPayload = null;
+
+        for (int task = 0; task < Integer.parseInt(tasksMax); task++) {
+            ObjectName objectName;
+            if (connector.equals("sql_server") || connector.equals("mongodb")) {
+                objectName = getObjectNameWithTask(connector, context, serverName, String.valueOf(task));
+            }
+            else {
+                objectName = getObjectName(connector, context, serverName);
+            }
+
+            Map<String, String> connectionAttributes = getAttributes(getConnectionAttributes(), objectName, connectorName, mBeanServer);
+            connectorPayload = new MetricsDescriptor.Connector(connectionAttributes);
+
+            List<MetricsDescriptor.Namespace> namespacesPayload = new ArrayList<>();
+            Map<String, String> connectorAttributes;
+            if (namespaces != null) {
+                for (String namespace : namespaces) {
+                    ObjectName objectNameWithNamespace = getObjectNameWithDatabase(connector, context, serverName, String.valueOf(task), namespace);
+                    connectorAttributes = getAttributes(getConnectorAttributes(), objectNameWithNamespace, connectorName, mBeanServer);
+                    namespacesPayload.add(new MetricsDescriptor.Namespace(namespace, connectorAttributes));
+                }
+            }
+            else {
+                connectorAttributes = getAttributes(getConnectorAttributes(), objectName, connectorName, mBeanServer);
+                namespacesPayload.add(new MetricsDescriptor.Namespace("", connectorAttributes));
+            }
+            tasksPayload.add(new MetricsDescriptor.Task(task, namespacesPayload));
+        }
+
+        return new MetricsDescriptor(connectorName, tasksMax, connectorPayload, tasksPayload);
+    }
+
+    @Override
+    default List<String> getConnectionAttributes() {
+        return CONNECTION_ATTRIBUTES;
+    }
+
+    @Override
+    default List<String> getConnectorAttributes() {
+        return CONNECTOR_ATTRIBUTES;
+    }
+
+    @GET
+    @Path(CONNECTOR_METRICS_ENDPOINT)
+    @Produces(MediaType.APPLICATION_JSON)
+    default MetricsDescriptor getConnectorMetrics(@PathParam("connector-name") String connectorName)
+            throws ReflectionException, MalformedObjectNameException, InstanceNotFoundException, IntrospectionException, AttributeNotFoundException, MBeanException {
+
+        if (getConnector() == null) {
+            throw new RuntimeException("Unable to fetch metrics for connector " + connectorName + " as the connector is not available.");
+        }
+
+        MBeanServer mBeanServer = ManagementFactory.getPlatformMBeanServer();
```

Comment:
This can be moved into a private const here in the interface and can be removed as parameter on the next line.